test/class_test.js

```diff
@@ -93,6 +93,67 @@ describe('classes', () => {
     `);
   });
 
+  describe('assign properties from method parameters', () => {
+    it('constructor without function body', () => {
+      check(`
+        class A
+          constructor: ([@a = 1], {test: @b = 2}, @c) ->
+      `, `
+        class A {
+          constructor([a = 1], {test: b = 2}, c) {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+          }
+        }
+      `);
+    });
+    it('constructor with function body', () => {
+      check(`
+        class A
+          constructor: (@a) ->
+            return
+      `, `
+        class A {
+          constructor(a) {
+            this.a = a;
+            return;
+          }
+        }
+      `);
+    });
+    it('method', () => {
+      check(`
+        class A
+          method: (@a) ->
+      `, `
+        class A {
+          method(a) {
+            this.a = a;
+            return;
+          }
+        }
+      `);
+    });
+    it('bound method', () => {
+      check(`
+        class A
+          method: (@a) =>
+      `, `
+        class A {
+          constructor() {
+            this.method = this.method.bind(this);
+          }
+        
+          method(a) {
+            this.a = a;
+            return;
```

Comment:
We often use literal `return` statements in coffee to avoid for example getting `for` loops as expressions. So I feel this a natural thing to do. On the other hand, we could generally drop `return` statements (without values) when they are the last statement in a function? Maybe it's not necessary to introduce such a magic identifier…